### ISSUE TITLE
Tests to enable open-source contributions

### ISSUE THREAD

**Issue by nickumia** [WaseemHussain/MRF-SDK#1]
Are there any tools or environments to deploy tests in so that the code can be more easily contributed to?  With robotic applications, hardware is always fun and nice to see it work; however, it gives a lot of peace of mind to know the algorithms are thoroughly tested in a sort of verification and validation mindset.

I used Gazebo a bit in older positions, but that's more for human-guided testing.  I think automated testing would be really nice.  Unit tests are a must.  But integration, regression and simulation testing can definitely be iterative.

**Comment by waseemhnyc** [comment]
@nickumia I completely agree! With the correct tests and simulation a developer could go from development to production much quicker. I think it would be great if there was a platform that included a hardware and software product that could make it easy to do all of this together. I know ROS attempts to do this but Gazebo imo isn't for the faint hearted 😅.
AWS also has AWS Robomaker which I use to teach ROS. I believe there are hardware integrations with Robomaker that helps as well.

I'll keep this issue until I have the time to add some of those tools in this repo.

**Comment by nickumia** [comment]
Do you still depend on ROS Kinetic?  Or are things upgradeable in the code?  Apparently Noetic is the [recommended](http://wiki.ros.org/Distributions) now.

I'll write a basic example of unit tests and start a PR to help coordinate.

**Comment by nickumia** [comment]
@waseemh-io Hey bro,  I started a PR #2 to add some of this capability.  I preemptively did a repo reorg (which you're free to reject).  If so, I can add the tests that I did and revert the file structure if it's too inhibitive to the original intent.

Let me know how you feel about the initial unit-level tests and I can continue doing them for the rest of the code.

**Comment by nickumia** [comment]
@waseemh-io friendly ping to get your thoughts.  I'm excited to set this up.

**Comment by waseemhnyc** [comment]
I am only like a whole year late with this 😵‍💫

**Comment by waseemhnyc** [comment]
@nickumia Think I'll be revising this later this year. Will leave this issue open until then :)

**Comment by nickumia** [comment]
Another year later 🙃
